src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -1766,4 +1749,247 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
+        }
+
+        // Re-assign to prompt reactivity
+        this.formState = JSON.parse(JSON.stringify(this.formState));
+    }
+
+    updateFormStateForDonationRecordType(fields) {
+        const opportunityRecordTypeValue = fields[DONATION_RECORD_TYPE_NAME.fieldApiName];
+
+        if (opportunityRecordTypeValue) {
+            const isId = opportunityRecordTypeValue.startsWith('012');
+            const val = isId ?
+                opportunityRecordTypeValue :
+                this.opportunityRecordTypeIdFor(opportunityRecordTypeValue);
+
+            this.formState[DONATION_RECORD_TYPE_NAME.fieldApiName] = this.opportunityRecordTypeNameFor(val);
+            this.setDonationRecordTypeIdInFormState(val);
+        }
+    }
+
+    opportunityRecordTypeNameFor(id) {
+        const found = this.opportunityRecordTypeInfos &&
+            this.opportunityRecordTypeInfos
+                .find(recordTypeInfo => recordTypeInfo.recordTypeId === id);
+
+        return found && found.name;
+    }
+
+    hasImportedRecordFieldsBeingSetToNull(fields) {
+        return Object.keys(fields)
+            .filter(field =>
+                GeFormService.importedRecordFieldNames.includes(field) &&
+                fields[field] === null
+            ).length > 0;
+    }
+
+    deleteRelationshipFieldsFromStateFor(fields) {
+        const needsRelationshipFieldDeleted = (field) =>
+            this.hasRelatedRecordFieldInFormState(field) &&
+            fields[field] === null;
+
+        Object.keys(fields)
+            .filter(needsRelationshipFieldDeleted)
+            .forEach(field => {
+                this.deleteFieldFromFormState(relatedRecordFieldNameFor(field));
+            });
+    }
+
+    deleteFieldFromFormState(field) {
+        delete this.formState[field];
+    }
+
+    hasRelatedRecordFieldInFormState(field) {
+        return this.formState.hasOwnProperty(relatedRecordFieldNameFor(field));
+    }
+
+    handleFormFieldChange(event) {
+        const value = event.detail.value;
+        const label = event.detail.label;
+        const sourceField = this.sourceFieldFor(event.detail.fieldMappingDevName);
+
+        let valueForFormState = value;
+        if (sourceField === DONATION_RECORD_TYPE_NAME.fieldApiName) {
+            valueForFormState = label;
+            this.setDonationRecordTypeIdInFormState(value);
+        } else if (sourceField === DATA_IMPORT_DONATION_DONOR_FIELD.fieldApiName) {
+            this.handleDonationDonorChange(value)
+        }
+        this.updateFormState({[sourceField]: valueForFormState});
+    }
+
+    sourceFieldFor(fieldMappingDevName) {
+        return GeFormService.fieldMappings[fieldMappingDevName].Source_Field_API_Name;
+    }
+
+    /*******************************************************************************
+     * @description Analyzes the sections property to get initial values and set them
+     * in the formState property.
+     */
+    initializeState() {
```

Comment:
Can we call this function `initializeFormState` so we're inline with `updateFormState` and `resetFormState`?

Reply:
Yep good call!